helpers/deploy.ts

```diff
@@ -15,8 +15,9 @@ export async function createDeploy(
 
   return async (contractName: string, params: string[] = []): Promise<[Contract, string]> => {
     const contractFactory = await ethers.getContractFactory(contractName, config.signer)
-    const instance = await contractFactory.deploy(...params)
+    const instance = await contractFactory.connect(config.signer).deploy(...params)
```

Comment:
Um you don't have to do that because ` config.signer` has been already provided  in `getContractFactory`.

Reply:
Have left a comment in the code.

```suggestion
    const contractFactory = await ethers.getContractFactory(contractName, {
      signer: await ethers.getSigner(await config.signer.getAddress()),
    })
```

Basically, the signer is not being persisted through when the deploy function is called - when using non-default signer